Jenkinsfile

```diff
@@ -1,52 +1,135 @@
-@Library('xmos_jenkins_shared_library@v0.14.2') _
+@Library('xmos_jenkins_shared_library@v0.15.0') _
 
 getApproval()
 
 pipeline {
-  agent {
-    label 'x86_64&&brew'
+  agent none
+  parameters {
+    string(
+      name: 'AI_TOOLS_VERSION',
+      defaultValue: '15.0.3',
+      description: 'The tools version to build with (check /projects/tools/ReleasesTools/)'
+    )
   }
-  environment {
-    REPO = 'lib_logging'
-    VIEW = getViewName(REPO)
-  }
-  options {
-    skipDefaultCheckout()
-  }
-  stages {
-    stage('Get view') {
-      steps {
-        xcorePrepareSandbox("${VIEW}", "${REPO}")
-      }
-    }
-    stage('Library checks') {
-      steps {
-        xcoreLibraryChecks("${REPO}")
-      }
-    }
-    stage('Tests') {
-      steps {
-        runXmostest("${REPO}", 'tests')
-      }
-    }
-    stage('xCORE builds') {
-      steps {
-        dir("${REPO}") {
-          xcoreAllAppsBuild('examples')
-          xcoreAllAppNotesBuild('examples')
-          dir("${REPO}") {
-            runXdoc('doc')
+  stages{
+    stage ('Full test'){
+      parallel {
+        stage('Standard Run') {
+          agent {
+            label 'x86_64&&brew'
+          }
+          environment {
+            REPO = 'lib_logging'
+            VIEW = getViewName(REPO)
+          }
+          options {
+            skipDefaultCheckout()
+          }
+          stages{
+            stage('Get view') {
+              steps {
+                sh 'rm -f cur_view.txt' //Temp hack. Need to work out why post/cleanup is not doing its job
+                xcorePrepareSandbox("${VIEW}", "${REPO}")
+              }
+            }
+            stage('Library checks') {
+              steps {
+                xcoreLibraryChecks("${REPO}")
+              }
+            }
+            stage('Tests') {
+              steps {
+                runXmostest("${REPO}", 'legacy_tests')
+              }
+            }
+            stage('xCORE builds') {
+              steps {
+                dir("${REPO}") {
+                  xcoreAllAppsBuild('examples')
+                  xcoreAllAppNotesBuild('examples')
+                  dir("${REPO}") {
+                    runXdoc('doc')
+                  }
+                }
+              }
+            }
+          }
+          post {
+            success {
+              updateViewfiles()
+            }
+            cleanup {
+              xcoreCleanSandbox()
+            }
           }
         }
-      }
-    }
-  }
-  post {
-    success {
-      updateViewfiles()
-    }
-    cleanup {
-      xcoreCleanSandbox()
-    }
-  }
+        stage('XCORE AI tests') {
+          agent {
+            label "srv-bri-nuc2"
```

Comment:
The labels "xs3" and "xcore.ai-explorer" are now available on all the relevant machines